app/src/main/java/nu/yona/app/ui/message/MessageStickyRecyclerAdapter.java

```diff
@@ -64,63 +71,107 @@ public MessageItemViewHolder onCreateViewHolder(ViewGroup parent, int viewType)
-					holder.img_avtar.setVisibility(View.GONE);
-					holder.profileIconTxt.setVisibility(View.VISIBLE);
-					holder.profileIconTxt.setText(yonaObject.getNickname().substring(0, 1).toUpperCase());
-					holder.profileIconTxt.setBackground(ContextCompat.getDrawable(YonaActivity.getActivity(), R.drawable.bg_small_self_round));
-				}
-			}
-			if (yonaObject.getLinks() != null && yonaObject.getLinks().getMarkRead() != null && !TextUtils.isEmpty(yonaObject.getLinks().getMarkRead().getHref()))
-			{
-				holder.messageContainer.setBackground(ContextCompat.getDrawable(activity, R.drawable.item_selected_gradient));
-			}
-			else
-			{
-				holder.messageContainer.setBackground(ContextCompat.getDrawable(activity, R.drawable.item_gradient));
-			}
-			holder.deleteMsg.setTag(yonaObject);
-			holder.messageContainer.setTag(yonaObject);
+			setUpNonSystemMessageItemHolder();
 		}
+		setUpMessageContainerBackground();
+	}
 
+	private void setUpSystemMessageItemHolder()
+	{
+		currentMessageItemHolder.img_avtar.setVisibility(View.GONE);
+		if (!TextUtils.isEmpty(currentYonaMessage.getMessage()))
+		{
+			currentMessageItemHolder.txtFooterMsg.setText(currentYonaMessage.getMessage());
+		}
+		currentMessageItemHolder.profileIconTxt.setVisibility(View.VISIBLE);
+		currentMessageItemHolder.profileIconTxt.setText(currentYonaMessage.getNickname().substring(0, 1).toUpperCase());
+		currentMessageItemHolder.profileIconTxt.setBackground(ContextCompat.getDrawable(YonaActivity.getActivity(), R.drawable.bg_small_admin_round));
+	}
+
+	private void setUpNonSystemMessageItemHolder()
+	{
+		currentMessageItemHolder.txtFooterMsg.setText(currentYonaMessage.getNickname());
+		if (currentYonaMessage.getNotificationMessageEnum() == NotificationMessageEnum.GOALCONFLICTMESSAGE_ANNOUNCED)
+		{
+			currentMessageItemHolder.img_avtar.setImageResource(R.drawable.adult_sad);
+			currentMessageItemHolder.img_avtar.setVisibility(View.VISIBLE);
+			currentMessageItemHolder.profileIconTxt.setVisibility(View.GONE);
+		}
+		else
+		{
+			setMessageListItemUserAvatar();
+		}
+	}
+
+	private void setMessageListItemUserAvatar()
+	{
+		if (currentYonaMessage.getLinks().getUserPhoto() != null)
+		{
+			Picasso.with(this.activity).load(currentYonaMessage.getLinks().getUserPhoto().getHref()).noFade().into(currentMessageItemHolder.img_avtar);
+			currentMessageItemHolder.img_avtar.setVisibility(View.VISIBLE);
+			currentMessageItemHolder.profileIconTxt.setVisibility(View.GONE);
+		}
+		else
+		{
+			currentMessageItemHolder.img_avtar.setVisibility(View.GONE);
+			currentMessageItemHolder.profileIconTxt.setVisibility(View.VISIBLE);
+			currentMessageItemHolder.profileIconTxt.setText(getMessageProfileIconText());
+			currentMessageItemHolder.profileIconTxt.setBackground(ContextCompat.getDrawable(YonaActivity.getActivity(), R.drawable.bg_small_self_round));
+		}
+	}
+
+	private String getMessageProfileIconText()
+	{
+		RegisterUser registerUser = currentYonaMessage.getEmbedded().getYonaUser();
```

Comment:
This is tricky. Not all messages have an embedded user. More often than not, the message contains a link to the user, which you then need to find among the buddies of the user.

Next, the name ``registerUser`` isn't very meaningful. Please make it ``senderUser``.

Reply:
what should I use when there is not ```yona:user``` present in the reponse.

Reply:
As far as I know, there is always either a ``yona:user`` link or an embedded user.

Reply:
👍

Reply:
You could add an enum to your notification enum type indicating one of the following three options:
* No user (for ``SystemMessage`` and for ``NONE_NONE``)
* Embedded user (``BuddyConnectRequest`` and ``BuddyDisconnect``)
* Linked user (others)

Reply:
Can I directly use Nickname associated with each notification object instead. As this is required field?

Reply:
Yes, you can use that. Sorry for making it more complex than necessary.

Reply:
Wait. The avatar is built from the first and last name, so the nickname is of no help here.

Reply:
> Avatar section https://wiki.yona.nu/display/DEV/General+Requirements 
discussion ends saying to use Nickname. Not sure if it is applicable to notifications also.

Reply:
Yes, it should be used everywhere. So then the nickname suffices (for buddies, for yourself it should be based on first and last name)

Reply:
In notifications logged in user will never have to be shown.

Reply:
They will: for goal conflicts of yourself.

Reply:
Ok. In that case ```yona:user``` should be used to derive first name and last name.

Reply:
:) Lucky you! You're right.